monitoring/perf_level.cc

```diff
@@ -4,25 +4,49 @@
 //  (found in the LICENSE.Apache file in the root directory).
 //
 
-#include <assert.h>
-#include "monitoring/perf_level_imp.h"
+#include "rocksdb/perf_level.h"
 
-namespace rocksdb {
+#include <cassert>
+#include <cstring>
+
+#include "monitoring/perf_flags_imp.h"
 
-#ifdef ROCKSDB_SUPPORT_THREAD_LOCAL
-__thread PerfLevel perf_level = kEnableCount;
-#else
-PerfLevel perf_level = kEnableCount;
-#endif
+namespace rocksdb {
 
 void SetPerfLevel(PerfLevel level) {
+  // TODO: discuss keep assertions or not
   assert(level > kUninitialized);
   assert(level < kOutOfBounds);
-  perf_level = level;
+  switch (level) {
+    case kEnableTime:
+      perf_flags = PERF_LEVEL5;
+      break;
+    case kEnableTimeAndCPUTimeExceptForMutex:
+      perf_flags = PERF_LEVEL4;
+      break;
+    case kEnableTimeExceptForMutex:
+      perf_flags = PERF_LEVEL3;
+      break;
+    case kEnableCount:
+      perf_flags = PERF_LEVEL2;
+      return;
+    default:
+      perf_flags = {};
+      break;
+  }
+  //  perf_flags.perf_level = level;
 }
-
 PerfLevel GetPerfLevel() {
-  return perf_level;
+  void* levels[5] = {(void*)&PERF_LEVEL1, (void*)&PERF_LEVEL2,
```

Comment:
I guess you can define PERF_LEVEL as a constant array, then you don't need to build the array here.